### PR TITLE
Code organization w/ sigs.k8s.io/release-utils

### PR DESCRIPTION
**Pull request by justaugustus**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Following conversations with @hasheddan and @saschagrunert yesterday, this is a path forward on breaking the repo out into more lightweight units for consumers.

Release imports will be disambiguated as follows:
- k8s.io/release: release CLI tooling, production stage/release configs and packages, and legacy imports
- [sigs.k8s.io/release-sdk](https://github.com/justaugustus/release-sdk): release interfaces and generic implementations
- [sigs.k8s.io/release-utils](https://github.com/justaugustus/release-utils): small, sharp packages with minimal imports

So the import graph for a Release Manager CLI tool may look like:

```console
sigs.k8s.io/release-utils --> sigs.k8s.io/release-sdk --> k8s.io/release/pkg
```

Whereas consumers, like kubetest2 and zeitgeist, may only need to import packages from sigs.k8s.io/release-utils or sigs.k8s.io/release-sdk.

This addresses a few discussions on code organization:
- https://github.com/kubernetes/release/issues/1819
- https://github.com/kubernetes/release/pull/1900
- https://github.com/kubernetes/release/pull/1908

/assign @saschagrunert @hasheddan 
cc: @kubernetes/release-engineering 

Initial packages to address:

- [x] (https://github.com/justaugustus/release-utils/pull/1) https://pkg.go.dev/k8s.io/release@v0.7.0/pkg/command?tab=importedby
- [x] (https://github.com/justaugustus/release-utils/pull/1) https://pkg.go.dev/k8s.io/release@v0.7.0/pkg/log?tab=importedby
- [x] (https://github.com/justaugustus/release-utils/pull/1) https://pkg.go.dev/k8s.io/release@v0.7.0/pkg/util?tab=importedby

Follow-up packages:

- [ ] https://pkg.go.dev/k8s.io/release@v0.7.0/pkg/build?tab=importedby
- [ ] https://pkg.go.dev/k8s.io/release@v0.7.0/pkg/git?tab=importedby
- [ ] https://pkg.go.dev/k8s.io/release@v0.7.0/pkg/github?tab=importedby
- [ ] https://pkg.go.dev/k8s.io/release@v0.7.0/pkg/notes?tab=importedby

#### Does this PR introduce a user-facing change?

```release-note
- The following packages have been migrated to sigs.k8s.io/release-utils:
  - pkg/command
  - pkg/editor
  - pkg/hash
  - pkg/http
  - pkg/log
  - pkg/tar
  - pkg/util
```

### CODE REVIEW

File: cmd/cip-mm/main.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ import (
 	"github.com/spf13/cobra"
 
 	reg "k8s.io/release/pkg/cip/dockerregistry"
-	"k8s.io/release/pkg/log"
+	"sigs.k8s.io/release-utils/log"
```
Comment (saschagrunert) [inline cmd/cip-mm/main.go:27]: What's the reason to use `sigs.k8s.io` over `k8s.io`?

Comment (justaugustus) [reply cmd/cip-mm/main.go:27]: @saschagrunert -- By default, new repositories would live under kubernetes-sigs with the vanity import `sigs.k8s.io`: https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md#rules-for-new-repositories

We could request [core repositories](https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md#core-repositories), but that also requires approval from SIG Architecture and I don't feel that it's strictly necessary for these repos.

Thoughts?

Comment (saschagrunert) [reply cmd/cip-mm/main.go:27]: Yep, I think then we can stick to sigs.k8s.io.

Comment (justaugustus) [reply cmd/cip-mm/main.go:27]: SGTM!

**Comment by justaugustus** [comment]
I've opened repo migration requests here: https://github.com/kubernetes/org/issues/2539

For this initial sweep, I've moved the following packages into release-utils:
- pkg/command
- pkg/editor
- pkg/hash
- pkg/http
- pkg/log
- pkg/tar
- pkg/util

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**